source/tests/system/nidigital_driver_api_tests.cpp

```diff
@@ -0,0 +1,292 @@
+    digital::GetAttributeViInt64Response response;
+    ::grpc::Status status = GetStub()->GetAttributeViInt64(&context, request, &response);
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+    return response.value();
+  }
+
+  std::string get_string_attribute(const char* channel_name, digital::NiDigitalAttributes attribute)
+  {
+    ::grpc::ClientContext context;
+    digital::GetAttributeViStringRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute(attribute);
+    digital::GetAttributeViStringResponse response;
+    ::grpc::Status status = GetStub()->GetAttributeViString(&context, request, &response);
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+    return response.value();
+  }
+
+ private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+  std::unique_ptr<digital::NiDigital::Stub> nidigital_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<digital::NiDigitalLibrary> nidigital_library_;
+  std::unique_ptr<digital::NiDigitalService> nidigital_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiDigitalDriverApiTest, SelfTest_SelfTestCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  digital::SelfTestRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  digital::SelfTestResponse response;
+  ::grpc::Status status = GetStub()->SelfTest(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+  EXPECT_EQ(0, response.test_result());
+  EXPECT_LT(0, strlen(response.test_message().c_str()));
+}
+
+TEST_F(NiDigitalDriverApiTest, Reset_ResetCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  digital::ResetRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  digital::ResetResponse response;
+  ::grpc::Status status = GetStub()->Reset(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiDigitalDriverApiTest, SetViInt32Attribute_GetViInt32Attribute_ValueMatchesSetValue)
+{
+  const char* channel_name = "";
+  const digital::NiDigitalAttributes attribute_to_set = digital::NiDigitalAttributes::NIDIGITAL_ATTRIBUTE_SELECTED_FUNCTION;
+  const ViInt32 expected_value = digital::SelectedFunction::SELECTED_FUNCTION_NIDIGITAL_VAL_PPMU;
+  ::grpc::ClientContext context;
+  digital::SetAttributeViInt32Request request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute(attribute_to_set);
+  request.set_value(expected_value);
+  digital::SetAttributeViInt32Response response;
+  ::grpc::Status status = GetStub()->SetAttributeViInt32(&context, request, &response);
+  ViInt32 get_attribute_value = get_int32_attribute(channel_name, attribute_to_set);
```

Comment:
Add a newline before this line so that test structure matches name of the test. Your test name contains 3 parts - "initial-setup", "action" and "validation". We should separate out our test code with newlines to arrange them in similar sections.

Same for rest of the tests below.